include/buildpack.bash

```diff
@@ -21,6 +21,54 @@ _envfile-parse() {
     done <<< "$(cat)"
 }
 
+_move-build-to-app() {
+	shopt -s dotglob nullglob
+	# shellcheck disable=SC2086
+	rm -rf ${app_path:?}/*
+	#  build_path defined in outer scope
+	# shellcheck disable=SC2086,SC2154
+	mv $build_path/* $app_path
+	shopt -u dotglob nullglob
+}
+
+_select-buildpack() {
+	if [[ -n "$BUILDPACK_URL" ]]; then
+		title "Fetching custom buildpack"
+		# buildpack_path defined in outer scope
+		# shellcheck disable=SC2154
+		selected_path="$buildpack_path/custom"
+		rm -rf "$selected_path"
+
+		IFS='#' read -r url commit <<< "$BUILDPACK_URL"
+		buildpack-install "$url" "$commit" custom &> /dev/null
+		# unprivileged_user & unprivileged_group defined in outer scope
+		# shellcheck disable=SC2154
+		chown -R "$unprivileged_user:$unprivileged_group" "$buildpack_path/custom"
+		selected_name="$(unprivileged "$selected_path/bin/detect" "$build_path" || true)"
```

Comment:
this sets off my variable scope alarm but the alternative is to return an array i guess?